### DISCUSSION TITLE
10 second connection timeouts in Home assistant containers

### DISCUSSION THREAD

**Discussion by habuild** [SBFspot/SBFspot#558] [Q&A]
Hi @sbf-

You popped into my homeassistant addon thread the other day. Thanks for that!
You pointed out the 10 second connection time outs.

Do you have any suggestions as to how to further troubleshoot this connection issue. It seems to be limited to NUC (homeassistant OS amd64) and possibly VM builds(although that's way beyond me), the same addons work for myself and other on pi32-64 builds. The user that I was troubleshooting this with most recently tried a few things, which added about 50 posts to the thread.
https://community.home-assistant.io/t/sbfspot-bluetooth-homeassistant-addon/402671/64

The bluetooth controller was definitely visible for them in the addon. It just seems to not being picked up by sbfspot. I can't find any helpful debug options (-v2 -d2 etc) that help with connection issues.

My Dockerfile is potentially missing an amd64 dependency or something. It compiles sbfspot from git, not using pre-compiled(it doesn't look like you have a linux-amd64 pre-compiled). I added a copy of the github builder action below.

Any suggestions/help would be greatly appreciated.

[build amd64 sbfspot addon.txt](https://github.com/SBFspot/SBFspot/files/9224149/build.amd64.sbfspot.addon.txt)

https://github.com/habuild/hassio-addons repo

**Comment by sbf-** [comment]
I didn't follow this thread lately so I was not aware of the activity, but it's not an easy one it looks...
I don't have NUC, nor docker, so I can't test it. In the past I tested on VirtualBox and there I had no issues.
Is the BT controller usable by other means?

**Comment by habuild** [comment]
I actually got it to work on a VMware VM using Debian11 with docker/HA supervised via my win10 machine. So looks like it is actually functional. So at least I can rule out my poor cut and copy programming and the Addon.

Seems to come back to the Bluetooth hardware and range issues. Part of the issue is the security of HA itself. Makes it hard to say if the BT controller is usable, Although Bluetooth just got a big update in HA so should be easier going forward.

It seems to be
short duration 10seconds < --- well out of range or a BT issue
60seconds just in range, close enough to see the inverter, but not transfer data.
120seconds or longer seems to be a conflict with the host device and Bluetooth controller.

I was more wondering if there was some debugging available during the connection phase. It's kind of difficult to guess at what different connection durations means. All the debugging/logging is for after the connection.

Thanks again!
